### PR TITLE
Updates for Stac-api beta 1 (core & search). Add a Stac Arrivals catalog.

### PR DESCRIPTION
**Pull request by jeremyh**
A few things newly-added to the root Stac document:

```json
 "conformsTo": [
        "https://api.stacspec.org/v1.0.0-beta.1/core",
        "https://api.stacspec.org/v1.0.0-beta.1/item-search"
    ]
```

Links
```json
        {
            "title": "Collections",
            "description": "All product collections",
            "rel": "children",
            "type": "application/json",
            "href": "http://127.0.0.1:8080/stac/collections"
        },
        {
            "title": "Arrivals",
            "description": "Most recently added items",
            "rel": "child",
            "type": "application/json",
            "href": "http://127.0.0.1:8080/stac/arrivals/items"
        },
        {
            "title": "Item Search",
            "rel": "search",
            "type": "application/json",
            "href": "http://127.0.0.1:8080/stac/search"
        },
```

Includes related updates & fixes.
 
The new Arrivals Stac catalog is also linked from the header of the Arrivals page:
(It's a Stac Feature Collection in reverse chronological order of being indexed into the ODC)

Screenshot from 2020-12-23 11-16-37

---

These changes include another small Postgres index addition, similar to the one in #250.

- It makes the default `/stac/search` endpoint load much faster. These searches happen across *all* collections, so will not currently use our collection-specific indexes.
- It can be applied by running `cubedash-gen --init` again, but is not mandatory.

### CODE REVIEW

**Comment by jeremyh** [comment]
Added a fix for #255 here, because it's dependent on these changes.

**Comment by jeremyh** [comment]
Ready for review, despite my meddlings since clicking that button.

**Review by whatnick — APPROVED** [review]

File: cubedash/templates/arrivals.html [hunk 1]
```diff
@@ -7,6 +7,10 @@
     <div class="panel highlight">
         <h2 class="followed lonesome">Recent Dataset Arrivals</h2>
         <span class="header-follow">over {{ period_length.days }} days</span>
+    <a href="{{ url_for('stac.arrivals') }}" class="badge header-badge">
+        As Stac Catalog
+        <i class="fa fa-file-text-o" aria-hidden="true"></i>
+    </a>
```
Comment (pindge) [inline cubedash/templates/arrivals.html:13]: indent more on this `<a>` tag?

Comment (whatnick) [reply cubedash/templates/arrivals.html:13]: leaving this for later 👍 have questions around static html/js minification for explorer we can revisit this with. Also possible improvements with CDN.

**Pull request merged**